### PR TITLE
Fix tracking of auto disabled apps in Updater

### LINKED ISSUE #15416: Full automatic app update option in setting Gui
Please implement an option that all apps (or just activatet) update themselves at a certain time...this would be very useful! Thanks :-)

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #15416

### CODE REVIEW

**Comment by georgehrke** [comment]
@nickvergessen I'm not too deep into `\OC_App`, so any feedback is welcome.

**Comment by georgehrke** [comment]
Would like to have some review / comment whether this is the fix we are looking for.
Afterwards we should add some tests to make sure it doesn't break again.

File: lib/private/Updater.php [hunk 1]
```diff
@@ -405,6 +413,7 @@ private function checkAppsRequirements() {
 					throw new \UnexpectedValueException('The files of the app "' . $app . '" were not correctly replaced before running the update');
 				}
 				\OC::$server->getAppManager()->disableApp($app);
+				$this->autoDisabledApps[] = $app;
```
Comment (nickvergessen) [inline lib/private/Updater.php]: Maybe instead we move the logic to the appmanager method with a bool flag.
So when someone adds another spot that disables an app, they are reminded about the option.

**Comment by nickvergessen** [comment]
Added a commit, I think that's the cleaner way to do it.

File: lib/private/Updater.php [hunk 2]
```diff
@@ -74,6 +74,13 @@ class Updater extends BasicEmitter {
 		4 => 'Fatal',
 	];
 
+	/**
+	 * List of appIds that have automatically been disabled during upgrade
+	 *
+	 * @var String[]
+	 */
+	private $autoDisabledApps = [];
```
Comment (georgehrke) [inline lib/private/Updater.php]: this member variable should be obsolete now

**Comment by kyrofa** [comment]
Thanks for fixing this, @georgehrke. I assume backporting to v16 is also part of the plan, here, but wanted to mention it anyway.

**Comment by nickvergessen** [comment]
Failing tests are 
> PHP  Fatal error:  Uncaught PDOException: SQLSTATE[HY000]: General error:  2006 MySQL server has gone away in  /drone/src/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:41

So time to rebase?

**Review by MorrisJobke — APPROVED** [review]
Code makes sense and all usages of disableApp that are affected are covered 👍

**Comment by MorrisJobke** [comment]
> So time to rebase?

Rebased and auto squashed.

**Comment by kyrofa** [comment]
Still have a few test failures, here.

**Comment by kyrofa** [comment]
Hey folks, what's the hold up here?

**Review by rullzer — APPROVED** [review]

**Comment by kyrofa** [comment]
> The backport to stable16 failed. Please do this backport manually.

Backport to stable16 is #16753.

**Pull request merged**